src/scripts/main.js

```diff
@@ -1 +1,109 @@
 'use strict';
+
+let rightClick;
+let leftClick;
+
+function makeDiv(className) {
+  const div = document.createElement('div');
+
+  div.setAttribute('data-qa', 'notification');
+  div.setAttribute('class', className);
+  document.body.appendChild(div);
+
+  return div;
+}
+
+const firstPromise = () => {
+  return new Promise((resolve, reject) => {
+    document.addEventListener(
+      'click',
+      () => {
+        leftClick = true;
+
+        resolve('First promise was resolved');
+      },
+      { once: true },
+    );
+
+    // eslint-disable-next-line prefer-promise-reject-errors
+    setTimeout(() => reject('First promise was rejected!'), 3000);
+  });
+};
+
+const secondPromise = () => {
+  return new Promise((resolve) => {
+    document.addEventListener(
+      'click',
+      () => {
+        leftClick = true;
+
+        resolve('Second promise was resolved');
+      },
+      { once: true },
+    );
+
+    document.addEventListener(
+      'contextmenu',
+      () => {
+        rightClick = true;
+
+        resolve('Second promise was resolved');
+      },
+      { once: true },
+    );
+  });
+};
+
+const thirdPromise = () => {
+  return new Promise((resolve) => {
+    document.addEventListener(
+      'click',
+      () => {
+        if (rightClick === true) {
+          rightClick = false;
+          resolve('Third promise was resolved');
+        }
+      },
+      { once: true },
+    );
+
+    document.addEventListener(
+      'contextmenu',
+      () => {
+        if (leftClick === true) {
+          leftClick = false;
+          resolve('Third promise was resolved');
```

Comment:
Similar to the previous comment, the 'thirdPromise' also resolves based on the 'leftClick' variable being true during a 'contextmenu' event. Ensure that the logic here is robust and accounts for all possible sequences of user interaction.